rust/gitxetcore/src/data_processing_v2.rs

```diff
@@ -409,6 +418,19 @@ impl PointerFileTranslatorV2 {
                                 data_ub,
                             ));
                             add_new_data = false;
+                        } else if let (Some(idx), cas_lg) = {
+                            let lg = self.cas_data.read().await;
+                            (lg.chunk_hash_map.get(&chunk.hash).map(|s| *s), lg)
+                        } {
+                            let (_, (data_lb, data_ub)) = cas_lg.chunks[idx];
+
+                            file_info.push(FileDataSequenceEntry::new(
+                                CHUNK_IN_COMMON_AGGREGATOR,
+                                n_bytes,
+                                data_lb,
+                                data_ub,
+                            ));
+                            add_new_data = false;
                         } else if !file_info.is_empty()
                             && file_info.last().unwrap().cas_hash == MerkleHash::default()
```

Comment:
And I think a similar optimization logic should be applied to `CHUNK_IN_COMMON_AGGREGATOR`

Reply:
Hmm. So we will have this optimization in three places: against a function local aggregator, against a struct global aggregator, and against a permanent shard. Maybe some refactoring in the future.